python_modules/dagster/dagster/_core/definitions/metadata/metadata_set.py

```diff
@@ -102,6 +110,11 @@ def _extract_value(cls, field_name: str, value: Any) -> Any:
         """Based on type annotation, potentially coerce the value to the expected type."""
         ...
 
+    @classmethod
```

Comment:
I wonder if we'll run into situations in the future where we'll want to do value remapping also (e.g. we end up changing how the value for a specific metadata set is represented)

Reply:
definitely possible! open to changing the APIs to support that in the future